test/src/androidTest/kotlin/io/realm/ImportTests.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.runtimeapi.RealmModule
+import io.realm.util.TestRealmFieldTypes
+import test.Sample
+import test.link.Child
+import test.link.Parent
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class ImportTests {
+
+    @RealmModule(Parent::class, Child::class, Sample::class)
+    class MySchema
+
+    lateinit var tmpDir: String
+    lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = Utils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = MySchema(), path = "$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        Utils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun importPrimitiveDefaults() {
+        val sample = Sample()
+        realm.beginTransaction()
+        realm.copyToRealm(sample)
+        realm.commitTransaction()
+
+        val managed = realm.objects(Sample::class)[0]
+        for (value in TestRealmFieldTypes.values()) {
```

Comment:
we should check for the size as well

Reply:
Added an `else ->` to error for uncatched types. What that was you were referring to?

Reply:
I meant that checking the size will guarantee that you went through all checks

Reply:
We dont have access to the actual count of the fields, so for now this is guaranteed from raising errors if there are types that does not have a switch case. But I put a comment on improving and test if all properties in `Sample` is actually tested when/if we have access to the number of fields.